### ISSUE TITLE
[Linux]: Error parsing JSON: X get_image failed: error 8 (73, 0, 1174) [Self-Operating Computer][Error] something went wrong :( [Self-Operating Computer][Error] AI response Failed take action after looking at the screenshot 

### ISSUE THREAD

**Issue by Binaryxz** [OthersideAI/self-operating-computer#80]
Linux issue in the moment of run and permissions. 

![image](https://github.com/OthersideAI/self-operating-computer/assets/141841395/56c5d6ca-0ac2-41ae-b99b-4ef855b6599d)

**Comment by shubhexists** [comment]
Hi @Binaryxz can you please write a task, something like " open chrome and navigate to github " . This is just to check if it works or not. As per my knowledge, normal "conversational messages" like hello haven't been tested /supported yet.

**Comment by shubhexists** [comment]
Similar #79 .

**Comment by fr0g-66723067** [comment]
This seems to be related to Pillow and ImageGrab on machines (possibly just virtual machines) that have Wayland enabled.

Here is the fix that worked for me on Ubuntu 22.04.3 LTS in a virtualbox.

https://stackoverflow.com/questions/75752576/pillow-imagegrab-grab-not-working-on-a-virtual-machine

```
# Open this file
/etc/gdm3/custom.conf
```
```
# Modify this value
WaylandEnable=false
```

then

```
# Restart gdm3
sudo systemctl restart gdm3
```

**Comment by michaelhhogue** [comment]
@fr0g-66723067 This seems to be a valid solution. 

@Binaryxz Note that X-server is required for Linux at this time. Wayland is not supported.
